network/address.go

```diff
@@ -70,6 +82,86 @@ func (a Address) NetworkAddress() string {
 	return vals[1]
 }
 
+// NetworkAddressResolved returns the network address of the address, but resolved.
+// That is: the hostname resolved and the port joined by a colon.
+// It returns an empty string if the address is not valid.
+func (a Address) NetworkAddressResolved() string {
```

Comment:
I'm not sure I understand the need for a distinct `NetworkAddressResolved` and `Resolve`: both are doing the resolution, one is only giving the ip address while the other gives the IP address and the port... Do we really need two distinct functions ? Which one are we going to use later ?  I find it confusing. 

Another question is whether `Resolve` should return another `Address` instead of a plain string ? Since `NetworkAddress` already has the purpose of extracting the ip address and `Port` the port, therefore kind of "removing" the need for the two distinct methods mentionned above.